src/averageAges.js

```diff
@@ -55,7 +59,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    (onlyWithSon ? person.sex === 'm' : true)
+    && people.find(mother => mother.name === person.mother)
+  );
+
+  const ageDifference = children.map(child => {
+    const personMother = people.find(mother => mother.name === child.mother);
+
+    return child.born - personMother.born;
+  });
+
+  const yearsSum = ageDifference.reduce((sum, years) =>
+    sum + years, 0);
+
+  return yearsSum / ageDifference.length;
+}
+
+function calculateAverageAge(array) {
+  const yearsSum = array.reduce((years, man) =>
+    years + (man.died - man.born), 0);
+
+  return yearsSum / array.length;
 }
```

Comment:
You didn't fix this function based on my last comment.

**This function should just add and divide. She should not know what is `item.born` and `item.died` are.**

This function must accept an array of numbers, not people

What happens if we do division by 0 in JS? Add condition to return and if you didn't have value in left side you will return 0. `|| 0`

```suggestion
function calculateAverageAge(array) {
  const yearsSum = array.reduce((sum, age) => (sum + age), 0);

  return yearsSum / array.length || 0;
}
```